recipes/kafka.rb

```diff
@@ -24,21 +24,29 @@
   version node['cdap']['version']
 end
 
-kafka_log_dir =
-  if node['cdap']['cdap_site'].key?('kafka.log.dir')
+if node['cdap']['version'].to_f >= 3.5 && node['cdap']['cdap_site'].key?('kafka.log.dir')
+  Chef::Log.warn('kafka.log.dir has been deprecated. Please use kafka.server.log.dirs instead.')
+end
+
+kafka_log_dirs =
+  if node['cdap']['cdap_site'].key?('kafka.server.log.dirs')
+    node['cdap']['cdap_site']['kafka.server.log.dirs']
+  elsif node['cdap']['version'].to_f < 3.5 && node['cdap']['cdap_site'].key?('kafka.log.dir')
```

Comment:
actually it's not valid... ``kafka.log.dir`` is ignored in ``3.5.0``.  I filed https://issues.cask.co/browse/CDAP-7179 for this.